src/css/framework.css

```diff
@@ -0,0 +1,248 @@
+/* --- Variables --- */
+:root {
+
+	/* gutters play a major role in grids,
+	many declorations in the layout depend on the value set for the grid  */
+	--gutter-value: 2em;
+
+	/* set max width for container */
+	--max-content-width: 1000px;
+}
+
+/* --- General reset  --- */
+
+/* set entire document for border box */
+*,
+*:after,
+*:before {
+	box-sizing: border-box;
+}
+
+body {
+	margin: 0;
+}
+
+main {
+	max-width: var(--max-content-width);
+	margin: 0 auto;
+}
+
+/* --- Gutters --- */
+.row-gutters {
+
+	/* negative margin on container to offset the
+	margins on the blocks at the edges */
+	margin: 0 calc(var(--gutter-value) / -2);
+}
+
+.row-gutters > .block {
+
+	/* half on the sides which will equal a full next to another block */
+	margin: 0 calc(var(--gutter-value) / 2) 0;
+}
+
+/* --- Padding for rows and blocks --- */
+
+/* padding for row only, doesn't apply to blocks */
+.row-pad-all {
+	padding: var(--gutter-value);
+}
+
+.row-pad-vertical {
+	padding-top: var(--gutter-value);
+	padding-bottom: var(--gutter-value);
+}
+
+.row-pad-horizontal {
+	padding-right: var(--gutter-value);
+	padding-left: var(--gutter-value);
+}
+
+.row-pad-top {
+	padding-top: var(--gutter-value);
+}
+
+.row-pad-bottom {
+	padding-bottom: var(--gutter-value);
+}
+
+/* pad all blocks in a row or just and individual block */
+.block-pad,
+.row-pad-children > .block {
+	padding: 1em;
+}
+
+/* remove padding from an individual block where all
+blocks in a row are set to be padded */
+.row-pad-children .block.block-pad-none {
+	padding: 0;
+}
+
+/* --- Align blocks --- */
+
+/* alignment of all blocks of a row */
+.row-align-top {
+	align-items: flex-start;
+}
+
+.row-align-bottom {
+	align-items: flex-end;
+}
+
+.row-align-center {
+	align-items: center;
+}
+
+/* alignment of an individual block */
+.block-align-top {
+	align-self: flex-start;
+}
+
+.block-align-bottom {
+	align-self: flex-end;
+}
+
+.block-align-center {
+	align-self: center;
+}
+
+/* --- Vertically justify content in block by row or by individual block --- */
+.row .block.block-justify-top,
```

Comment:
Why is the specificity of the `.block` class needed here vs `.row .block-justify-top`? I already forgot. :)